assets/src/components/v2/departures/section.ts

```diff
@@ -0,0 +1,121 @@
+import _ from "lodash";
+
+import NormalSection, { Layout } from "./normal_section";
+import NoticeSection from "./notice_section";
+
+export type Section =
+  | (NormalSection & { type: "normal_section" })
+  | (NoticeSection & { type: "notice_section" });
+
+/**
+ * Perform one iteration of "section trimming", i.e. the process of removing
+ * departures from sections while respecting their desired layouts until they
+ * all fit in a container without overflow.
+ *
+ * Operates immutably. If no sections could be trimmed, returns the same array.
+ */
+export const trimSections = (sections: Section[]): Section[] => {
+  const trimmed = _.cloneDeep(sections);
+
+  const sortedSizedSections = trimmed
+    .map((section) => ({ section, length: sectionLength(section) }))
+    .sort(({ length: a }, { length: b }) => b - a);
+
+  for (const key in trimStages) {
+    if (trimStages[key](sortedSizedSections)) return trimmed;
+  }
```

Comment:
**question (non-blocking):** are we able to use `Object.values` here?

```suggestion
  for (const trimStage of Object.values(trimStages)) {
    if (trimStage(sortedSizedSections)) return trimmed;
  }
```

Reply:
Definitely. I think it was this way originally because I was logging `key` itself for debugging. Technically `trimStages` could just be an array instead of an object, but I do like the ability for the functions to have names.